FileSystemManager/Controllers/FileSystemController.cs

```diff
@@ -287,162 +289,249 @@ public IActionResult OpenFile(string path)
+                    if (fileMetadata != null)
+                    {
+                        fileMetadata.IsDeleted = true;
+                        fileMetadata.DeletedPath = recycleBinPath;
+                        fileMetadata.DateModified = DateTime.UtcNow;
+                        _context.Update(fileMetadata);
+                        await _context.SaveChangesAsync();
+                    }
+                    await transaction.CommitAsync();
+                    return Ok();
                 }
-                else
+                catch (Exception ex)
                 {
-                    return NotFound();
+                    await transaction.RollbackAsync();
+                    return StatusCode(500, "Error deleting item: " + ex.Message);
                 }
+            }
+        }
+        [HttpGet("RecycleBin")]
+        public IActionResult RecycleBin()
+        {
+            return View();
+        }
+        [HttpGet("FileSystem/RecycleBinItems")]
+        public async Task<IActionResult> GetRecycleBinItems()
+        {
+            try
+            {
+                var recycleBinItems = _context.FileMetadata
+                    .Where(fm => fm.IsDeleted)
+                    .ToList();
 
-                return Ok();
+                return Json(recycleBinItems);
             }
-            catch
+            catch (Exception ex)
             {
-                return StatusCode(500, "Error deleting item");
+                return StatusCode(500, "Error retrieving recycle bin items: " + ex.Message);
             }
         }
 
-        [HttpPost]
-        public IActionResult RenameItem(string path, string newName)
+        
+        [HttpPut("FileSystem/RestoreItem/{fileId}")]
+        public IActionResult RestoreItem(int fileId)
         {
-            if (string.IsNullOrEmpty(path) || string.IsNullOrEmpty(newName))
-            {
-                return BadRequest("Invalid path or new name");
-            }
-
-            var decodedPath = System.Net.WebUtility.UrlDecode(path);
-            var fullPath = Path.Combine(rootPath, decodedPath.TrimStart('/'));
-            var newFullPath = Path.Combine(Path.GetDirectoryName(fullPath), newName);
-
             try
             {
-                if (Directory.Exists(fullPath))
+                var fileMetadata = _context.FileMetadata.Find(fileId);
+                if (fileMetadata == null)
                 {
-                    Directory.Move(fullPath, newFullPath);
+                    return NotFound("File not found in recycle bin.");
                 }
-                else if (System.IO.File.Exists(fullPath))
+
+                string deletedPath = fileMetadata.DeletedPath; // Path where the file was originally deleted from
+                string restorePath = DetermineRestorePath(fileMetadata.FilePath); // Determine where to restore the file
+
+                if (System.IO.File.Exists(deletedPath))
```

Comment:
If this is not found here locally, it was still found in DB. What should be done about this inconsistency?

Reply:
Added

Reply:
What about directory?